tsfresh/feature_extraction/extraction.py

```diff
@@ -313,16 +314,18 @@ def _do_extraction(df, column_id, column_value, column_kind,
                                     function_kwargs=kwargs)
     distributor.close()
 
-    # Return a dataframe in the typical form (id as index and feature names as columns)
-    result = pd.DataFrame(result)
-    if "value" in result.columns:
-        result["value"] = result["value"].astype(float)
+    return_df = pivot_list(result)
 
-    if len(result) != 0:
-        result = result.pivot("id", "variable", "value")
-        result.index = result.index.astype(df[column_id].dtype)
+    # All the features are floats
+    return_df = return_df.astype(float)
```

Comment:
astype is a hotspot in your profile, you could remove this line if you add `dtype=float` to DataFrame constructor in `pivot_list`

Reply:
Nice catch. Actually I thought that this line does basically nothing in most of the cases. But yes, I should have been more careful.
Changed that!